usagov/usagov-2021
# USAGOV-1674 GTM category tag

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1674

## Description

Adds benefit categories to the datalayer for Google Tag Manager for analysis
## Type of Changes

- [x] Frontend (Twig, Sass, JS) Techincally, it's a twig update but there should be no visual changes. The source for the page changes:

![image](https://github.com/user-attachments/assets/b3f138a9-5475-49b1-8570-771314b6ef19)


## Testing Instructions


Setup the benefits search with some tagged pages. You should see one or two new values in the datalayer for standard pages.
### Change Requirements



### Validation Steps

- View source for a standard page that's been tagged to one or more benefit categories. The datalayer (script tag with id=taxonomy-data)  should have entries that look like the following. Categories should be separated by a `;`
```
      "hasBenefitCategory": true,
      "benefitCategories": "Housing; Utilities",
```
- View source for a standard page that has not been tagged. The datalayer should only have one entry

```
      "hasBenefitCategory": false,
```
- All other pages should not include either of these json keys.

## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
